src/main/resources/templates/placeOrder.html

```diff
@@ -17,121 +17,121 @@
+                        <option value="NJ">New Jersey</option>
+                        <option value="NM">New Mexico</option>
+                        <option value="NY">New York</option>
+                        <option value="NC">North Carolina</option>
+                        <option value="ND">North Dakota</option>
+                        <option value="OH">Ohio</option>
+                        <option value="OK">Oklahoma</option>
+                        <option value="OR">Oregon</option>
+                        <option value="PA">Pennsylvania</option>
+                        <option value="RI">Rhode Island</option>
+                        <option value="SC">South Carolina</option>
+                        <option value="SD">South Dakota</option>
+                        <option value="TN">Tennessee</option>
+                        <option value="TX">Texas</option>
+                        <option value="UT">Utah</option>
+                        <option value="VT">Vermont</option>
+                        <option value="VA">Virginia</option>
+                        <option value="WA">Washington</option>
+                        <option value="WV">West Virginia</option>
+                        <option value="WI">Wisconsin</option>
+                        <option value="WY">Wyoming</option>
+                    </select>
+                </div>
+                <div class="form-group col-md-2">
+                    <label for="inputZip">Zip</label>
+                    <input type="text" class="form-control" id="inputZip" th:field="*{customerZip}">
+                </div>
             </div>
-        </div>
-        <div class="form-row">
-            <div class="form-group col-md-6">
-                <label for="itemNumber">Add Item Number</label>
-                <input type="text" class="form-control" id="itemNumber" placeholder="Item number">
+            <div class="form-row">
+                <div class="form-group col-md-6">
+                    <label for="itemNumber">Add Item Number</label>
+                    <input type="text" class="form-control" id="itemNumber" placeholder="Item number" th:field="*{itemId}">
+                </div>
+                <div class="form-group col-md-6">
+                    <label for="quantity">Enter Quantity</label>
+                    <input type="text" class="form-control" id="quantity" placeholder="Quantity" th:field="*{quantity}">
+                </div>
             </div>
-            <div class="form-group col-md-6">
-                <label for="quantity">Enter Quantity</label>
-                <input type="text" class="form-control" id="quantity" placeholder="Quantity">
+            <div class="form-row">
+                <div class="form-group col-md-6">
+                    <label for="inlineFormCustomSelectPref">Shipping Vendor</label>
+                    <select class="custom-select my-1 mr-sm-2" id="inlineFormCustomSelectPref" th:field="*{carrier}">
```

Comment:
Added the field attribute to the input element. This will allow the element values to bind to the field of the object.